src/http/headers.cr

```diff
@@ -102,31 +102,25 @@ struct HTTP::Headers
   end
 
   def add(key, value : String)
-    check_invalid_header_content value
-
-    key = wrap(key)
-    existing = @hash[key]?
-    if existing
-      existing << value
-    else
-      @hash[key] = [value]
-    end
-    self
+    add(key, [value])
   end
 
   def add(key, value : Array(String))
     value.each { |val| check_invalid_header_content val }
-
-    key = wrap(key)
-    existing = @hash[key]?
-    if existing
-      existing.concat value
-    else
-      @hash[key] = value
-    end
+    unsafe_add(key, value)
     self
   end
 
+  def add?(key, value : String)
+    add?(key, [value])
```

Comment:
This allocates an array. The same for `add(key, String)`. I prefer it not to allocate (like before, have different implementations of the method)

Reply:
Or maybe we can just allow to pass a tuple here?

Reply:
@asterite updated.
Any other suggestions?